### PR TITLE
[CHAT-2693] Add flag reports api

### PR DESCRIPTION
**Pull request by ffenix113**
Added functionality to retrieve/review flag reports.

### CODE REVIEW

File: stream_chat/async_chat/client.py [hunk 1]
```diff
@@ -190,6 +190,29 @@ async def unflag_user(self, target_id: str, **options: Any) -> StreamResponse:
         data = {"target_user_id": target_id, **options}
         return await self.post("moderation/unflag", data=data)
 
+    async def query_flag_reports(self, **options: Any) -> StreamResponse:
```
[the remark this answers is not in the document]
Comment (gumuz) [reply stream_chat/async_chat/client.py:193]: looks like the Base class has return type signatures like `Union[StreamResponse, Awaitable[StreamResponse]]:`

[the remark this answers is not in the document]
Comment (gumuz) [reply stream_chat/async_chat/client.py:193]: I see you already fixed it :) @ffenix113

File: stream_chat/async_chat/client.py [hunk 2]
```diff
@@ -190,6 +190,29 @@ async def unflag_user(self, target_id: str, **options: Any) -> StreamResponse:
...
+        It is present for internal usage only.
+        This function can, and will, break and/or be removed at any point in time.
+        """
+        return await self.post("moderation/reports", data=options)
+
+    async def review_flag_report(
+        self, report_id: int, review_result: str, user_id: str, **details: Any
+    ) -> StreamResponse:
+        """
+        Note: Do not use this.
```
Comment (gumuz) [inline stream_chat/async_chat/client.py:205]: maybe prefix this method with an underscore to signify it is private

Comment (ffenix113) [reply stream_chat/async_chat/client.py:205]: Updated, please check.

**Comment by ffenix113** [comment]
Unfortunately tests will fail until backend is updated. Let's wait for now.

**Review by AnatolyRugalev — APPROVED** [review]

**Review by AnatolyRugalev — APPROVED** [review]

**Review by gumuz — APPROVED** [review]
LGTM

**Pull request merged**